clusterloader2/pkg/measurement/common/api_availability_measurement.go

```diff
@@ -0,0 +1,248 @@
+	wg                  sync.WaitGroup
+}
+
+func (a *apiAvailabilityMeasurement) updateMasterAvailabilityMetrics(c clientset.Interface, config *measurement.MeasurementConfig) {
+	for _, host := range a.hosts {
+		// update the number of probes
+		a.hostLevelMetrics[host].numProbes++
+		// SSH and check the health of the host
+		command := fmt.Sprintf("curl -s -k %slocalhost:%v/healthz", "https://", 443)
+		_, err := measurementutil.SSH(command, host+":22", "")
+		if err != nil {
+			a.updateAvailabilityMetrics(false, a.hostLevelMetrics[host])
+		} else {
+			a.updateAvailabilityMetrics(true, a.hostLevelMetrics[host])
+		}
+	}
+}
+
+func (a *apiAvailabilityMeasurement) updateClusterAvailabilityMetrics(c clientset.Interface) {
+	// Always update the Probes count
+	a.clusterLevelMetrics.numProbes++
+	// Check the availability of the cluster by issuing a REST call to /healthz end point
+	_, err := c.CoreV1().RESTClient().Get().AbsPath("/healthz").DoRaw()
+	if err != nil {
+		a.updateAvailabilityMetrics(false, a.clusterLevelMetrics)
+	} else {
+		a.updateAvailabilityMetrics(true, a.clusterLevelMetrics)
+	}
+}
+
+func (a *apiAvailabilityMeasurement) start(config *measurement.MeasurementConfig, SSHToMasterSupported bool, probeDuration int) ([]measurement.Summary, error) {
+	a.hosts = config.ClusterFramework.GetClusterConfig().MasterIPs
+	if len(a.hosts) < 1 {
+		return nil, goerrors.Errorf("Api Server measurement can't execute due to no MasterIps")
+	}
+
+	k8sClient := config.ClusterFramework.GetClientSets().GetClient()
+	a.isRunning = true
+	a.stopCh = make(chan struct{})
+	a.wg.Add(1)
+	go func() {
+		defer a.wg.Done()
+		for {
+			select {
+			case <-a.stopCh:
+				return
+			case <-time.After(time.Duration(probeDuration)):
+				a.updateClusterAvailabilityMetrics(k8sClient)
+				if SSHToMasterSupported {
+					a.updateMasterAvailabilityMetrics(k8sClient, config)
+				}
+				return
+			}
+
+		}
+	}()
+	return nil, nil
+}
+
+func (a *apiAvailabilityMeasurement) updateAvailabilityMetrics(apiServerAvailable bool, metrics *apiAvailabilityMetrics) {
+	if apiServerAvailable {
+		metrics.updateSuccessMetrics()
+	} else {
+		metrics.updateFailureMetrics()
+	}
+}
+
+// If the consecutive failed probes counter value > 0, then it means we are in the middle of an unavailability period
+// for the api server
+func (a *apiAvailabilityMetrics) previousProbeFailure() bool {
+	return a.consecutiveFailedProbes > 0
+}
+
+// Check if the recently concluded consecutive failed number of probes is
+// higher than the max consecutive failed number of probes so far
+// if yes, then Update max consecutive failed probes
+func (a *apiAvailabilityMetrics) mayBeUpdateMaxConsecutiveFailures() {
+	if a.consecutiveFailedProbes > a.maxConsecutiveFailedProbes {
+		a.maxConsecutiveFailedProbes = a.consecutiveFailedProbes
+	}
+}
+
+func (a *apiAvailabilityMetrics) updateFailureMetrics() {
+	// increment total number of failed probes count, current number of failed probes by 1
```

Comment:
remove this comment - don't describe what the code is doing - only describe why if that isn't obvious (in this case it is, so no comment is needed)